src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/SymbolTable.cs

```diff
@@ -106,18 +106,14 @@ private void ClearCache()
             {
                 callingType = callingType.GetGenericTypeDefinition();
             }
+
             if (name == SpecialNames.Indexer)
             {
-                // What about named indexers?
-                if (callingType == typeof(string))
-                {
-                    name = "Chars";
-                }
-                else
-                {
-                    name = "Item";
-                }
+                // If we don't find an indexer name for this type, use SpecialNames.Indexer as a key on the
+                // empty results we'll get, so that that gets cached.
+                name = callingType.GetIndexerName() ?? SpecialNames.Indexer;
```

Comment:
It doesn't need to be distinguished, because `(typeof(string).GetCustomAttribute(typeof(DefaultMemberAttribute)) as DefaultMemberAttribute)?.MemberName == "Chars"`. String is the most prominent type where it isn't "Item", so in not looking up the indexer and just assuming it would be "Item" it had to special-case it, but looking it up means we don't need to special case anything.